chall05/sadawi.rb

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+#ruby 2.5.1p57
+
+require 'open-uri'
+begin
+file = open('https://en.wikipedia.org/wiki/' + ARGV[0])
+rescue
+	puts "no wikipedia page for this word!"
+	exit
+end
+contents = file.read
+main = contents.match(/(?<=<p>)(?<!<\/p>)[\s\S]*?((?<=wiki\/)[a-zA-Z_()]*?(?="))[\s\S]*?<\/a>/)[1]
+visited = []
+count = 1
+puts "	Going to " + main + " (counter: " + count.to_s + ")"
+visited.push(main)
+
+while main != "Philosophy"
+	count += 1
+	file = open('https://en.wikipedia.org/wiki/' + main)
+	contents = file.read
+	main = contents.match(/(?<=<p>)(?<!<\/p>)[\s\S]*?((?<=wiki\/)[a-zA-Z_()]*?(?="))[\s\S]*?<\/a>/)[1]
+	if visited.include? main
```

Comment:
even if ruby allows to pass parameters that way I highly recommend keeping writing the other way.